src/tutorials/lua_basics_lightbulb.md

```diff
@@ -25,103 +25,184 @@ This involves turning on and off a light switch to illuminate a light bulb.
+
+### Make the Trigger Interactable
 
-## Writing the Code
+!["Interactable"](../img/LightBulb/image12.png){: .image-frame .center loading="lazy" }
 
-### Define the Switch
+1. Look at the **Properties** window with the trigger selected. Under the **Gameplay** section there is a property called **Interactable**. Make sure the box is checked so players can interact with the trigger.
+2. Find the **Interaction Label** property, and change it to something like "Turn On"
+3. Press **Play** to start a preview, and walk up to the Light switch. You should see a message pop up in the middle of your screen.
+
+## Setting the Custom Properties
+
+A very powerful feature of creating in Core is custom properties; they allow you to point reference to other objects or other data types, such as strings, numbers, etc.
+
+Thinking ahead before we begin scripting, we need to decide what custom properties we need.
+
+Being this is a lightbulb and switch, we know those are two custom properties we'll need already.
+
+Additionally, in order to be able to allow the player's interaction to cause the switch to flip and the light to turn on, we'll need a reference to the trigger we just customized above.
+
+To add a custom property to the script, click on the script and then drag-and-drop the object we are referencing into the **Custom** section of the **Properties** window.
+
+It should now look like this:
+!["Custom Properties"](../img/LightBulb/CustomProperties.png){: .image-frame .center loading="lazy" }
+
+## Scripting References to Custom Properties
+
+### Reference the Switch
 
 We want our light switch to function just like a real one: the switch will point up or down depending on whether the light is turned on or off.
 
 First you will need to tell the script which object in the scene is the switch, so that it knows what to rotate. You will create a variable that defines what the switch is.
 
 To move the switch, we need a way to talk about the specific object from the game space in code.
 
-Type the following into Line 1 of `LightToggleScript`:
+Type the following into Line 1 of `LightSwitchToggle`:
+
+```lua
+local Switch = script:GetCustomProperty("Switch"):WaitForObject()
+```
+
+- `local` is a **keyword** that indicates that you are making a variable. We use the term `local` here because it will only be accessible from this script rather than being accessible from external scripts, or globally.
+- `Switch` is the variable name. You can name it anything but it's important to create variables with self-explanatory names so our scripts are easy to read and understand.
+- `script` refers to the script itself as it exists in the project.
+- `GetCustomProperty(string name)` is a function that gets a reference to the object in the custom property that is a part of the script.
+- `WaitForObject()` is a function that is a part of an `CoreObjectReference` that returns the actual object as it exists in the **Main Viewport** and **Hierarchy**.
```

Comment:
This doesn't quite explain why it's waiting, CoreObjectReference isn't super meaningful, or very clear what would be different from it and CoreObject just by name.

Emphasize:
- this is an object that is actually in the game and Hierarchy
- it prevents the script from trying to find it before it has been loaded